components/binding-usage-controller/internal/controller/usagekind/protection_controller_test.go

```diff
@@ -0,0 +1,143 @@
+package usagekind_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kyma-project/kyma/components/binding-usage-controller/internal/controller/usagekind"
+	"github.com/kyma-project/kyma/components/binding-usage-controller/internal/platform/logger/spy"
+	"github.com/kyma-project/kyma/components/binding-usage-controller/pkg/apis/servicecatalog/v1alpha1"
+	"github.com/kyma-project/kyma/components/binding-usage-controller/pkg/client/clientset/versioned/fake"
+	informers "github.com/kyma-project/kyma/components/binding-usage-controller/pkg/client/informers/externalversions"
+)
+
+func TestProtectionControllerAddsFinalizer(t *testing.T) {
+	// GIVEN
+	tc := newProtectionControllerTestCase()
+	defer tc.TearDown()
+	tc.RunController()
+
+	// WHEN
+	tc.AddUsageKind(t, fixUsageKind())
+	tc.WaitForAddFinalizer(t, 3*time.Second)
+
+	// THEN
+	uk := tc.GetUsageKind(t, fixUsageKind().Name)
+	assert.Contains(t, uk.Finalizers, usagekind.FinalizerName)
+
+	// WHEN
+	tc.MarkRemoval(t, uk.Name)
+	tc.WaitForProcessDeletion(t, 3*time.Second)
```

Comment:
IMO it's more like `tc.WaitUnitlDeletionProcessDone`
because you are not waiting for triggering the deletion process, but you are waiting unit it's finished